### ISSUE TITLE
[restartless/e10s] Search Engine opening link "redirect" empty background tab

### ISSUE THREAD

**Issue by ldgbc** [RequestPolicyContinued/requestpolicy#577]
Firefox 35
RP: https://github.com/RequestPolicyContinued/requestpolicy/commit/beafc6076947096d5528c6875f27f6a21da1d2a0

Test URL: https://startpage.com/do/search
Search: RequestPolicy

Problem: Using "Middle Click" on a link in search engine will open it in a new tab, however RP will block the request thinking it is a redirect. It will pop up asking "Allow, Deny, Add Rule"

**Comment by myrdd** [comment]
Wait, this is not a bug. At least in my case it tries to redirect to a different domain (`*.com` to `*.de`). Please re-check.

```
origin: https://www.google.com/search?q=bbc&ie=utf-8&oe=utf-8
destination: https://www.google.de/search?q=bbc&ie=utf-8&oe=utf-8&gws_rd=cr&e
```

However, clicking "Allow" does not work.

**Comment by ldgbc** [comment]
Try it with Bing.com or StartPage.com
Google don't seem to have this problem because it just it own redirect

It would seem that this is not just search engine result. Middle Clicking on many URL is getting the popup too. 
For example: Clicking on "

Test URL: https://startpage.com/do/search" that I linked in the first comment would also pop this up.

![untitled](https://cloud.githubusercontent.com/assets/1805303/5793911/ba8066b4-9fab-11e4-8171-6bdf7491e648.jpg)

Once I choose "Allow", it will load in current tab, clicking "Back", and opening it with Middle Click will make it work after.

![untitled2](https://cloud.githubusercontent.com/assets/1805303/5793912/fe781ace-9fab-11e4-82f1-546ee14c8bf3.jpg)

**Comment by myrdd** [comment]
> Test URL: https://startpage.com/do/search

sorry, I've missed that link

I can confirm that middle-clicking on links leads to a buggy behaviour. However, middle-clicking on any search engine works for me:

![bing](https://cloud.githubusercontent.com/assets/4286829/5793987/e2982980-9f5a-11e4-98ee-c36dad7b70a7.jpg)

Which of your settings are different from the default ones? — Default prefs are marked bold on `about:config`. RP prefs are starting with `extensions.requestpolicy`. In my case it's only `defaultPolicy.allow` being `false`.

**Comment by ldgbc** [comment]
I understand why you not reproducing result. You just doing a search using the search bar, whereas I meant when opening the "search result". See video

https://mediacru.sh/S2pymYiYqx6d

As you can see the 1st link work fine, it open in the 2nd tab and the link is highlighted meaning it is in my browser history.
The 2nd and 3rd link open blank tab, and RP popup with the "Allow, Deny, Add Rule" offer.

**Comment by nodiscc** [comment]
For what it's worth, I can't reproduce this on the `dev-1.0` branch.

**Comment by myrdd** [comment]
@nodiscc this is on `dev-restartless-and-e10s`, see the title.

@ldgbc thank you for the video, now I got it! I can reproduce this bug.

**Comment by myrdd** [comment]
The bug described is also reproducible when e10s is disabled. It seems like the middle-click is not registered as a link click. I will work on this.

**Comment by myrdd** [comment]
this bug is fixed in commit 521ce64. Thank you for reporting and explaining it to me!
